src/nomad_simulations/outputs.py

```diff
@@ -17,89 +17,214 @@
+        super().__init__(m_def, m_context, **kwargs)
+        self.name = self.m_def.name
+
+
+class Energies(Variables):
+    def __init__(self, m_def: Section = None, m_context: Context = None, **kwargs):
+        super().__init__(m_def, m_context, **kwargs)
+        self.name = self.m_def.name
+
+
+class PhysicalProperty(ArchiveSection):
+    """ """
+
+    source = Quantity(
+        type=MEnum('simulation', 'measurement', 'analysis'),
+        default='simulation',
+        description="""
+        Source of the physical property. Example: an `ElectronicBandGap` can be obtained from a `'simulation'`
+        or in an `'measurement'`.
+        """,
+    )
+
+    type = Quantity(
         type=str,
         description="""
-        Name of the output property. This is used for easier identification of the property and is connected
-        with the class name of each output property class, e.g., `'ElectronicBandGap'`, `'ElectronicBandStructure'`, etc.
+        Type categorization of the physical property. Example: an `ElectronicBandGap` can be `'direct'`
+        or `'indirect'`.
         """,
-        a_eln=ELNAnnotation(component='StringEditQuantity'),
     )
 
-    orbitals_state_ref = Quantity(
-        type=OrbitalsState,
+    label = Quantity(
+        type=str,
         description="""
-        Reference to the `OrbitalsState` section to which the output property references to and on
-        on which the simulation is performed.
+        Label for additional classification of the physical property. Example: an `ElectronicBandGap`
+        can be labeled as `'DFT'` or `'GW'` depending on the methodology used to calculate it.
         """,
-        a_eln=ELNAnnotation(component='ReferenceEditQuantity'),
     )
 
-    atoms_state_ref = Quantity(
-        type=AtomsState,
+    shape = DirectQuantity(
+        type=Dimension,
+        shape=['0..*'],
+        default=[],
+        name='shape',
         description="""
-        Reference to the `AtomsState` section to which the output property references to and on
-        on which the simulation is performed.
+        Shape of the physical property. This quantity is related with the order of the tensor which
+        describes the physical property:
+            - scalars (tensor order 0) have `shape=[]` (`len(shape) = 0`),
+            - vectors (tensor order 1) have `shape=[a]` (`len(shape) = 1`),
+            - matrices (tensor order 2), have `shape=[a, b]` (`len(shape) = 2`),
+            - etc.
         """,
-        a_eln=ELNAnnotation(component='ReferenceEditQuantity'),
     )
 
-    model_system_ref = Quantity(
-        type=ModelSystem,
+    variables = SubSection(
+        type=Variables.m_def,
         description="""
-        Reference to the `ModelSystem` section to which the output property references to and on
-        on which the simulation is performed.
+        Variables over which the physical property varies. These are defined as binned, i.e., discretized
```

Comment:
Ok, I see. Quickly answering to the fact that bins can be anything, then I'd suggest go with `Any` as type, meaning, it can be any booleans, integers, floats. I am not 100% sure this is the correct typing, but let me ask.

Reply:
I added a couple of warning comments (`# !`) for adding more examples in `label` and `type` in a future.